### ISSUE TITLE
Compiler Segfaults without `--obj-out` specified

### ISSUE THREAD

**Issue by Caleb-o** [c3lang/c3c#1503]
# Issue
I'm trying to compile [my project](https://github.com/Caleb-o/c3-with-c) directly with the compiler and I hit a segault. It seems that `--obj-out` must be specified for it to work.

### Command in question:
`c3c compile src/main.c3 --output-dir temp --libdir lib --lib littlec -o helloc` will cause a segfault:

![image](https://github.com/user-attachments/assets/e0d26429-145a-441f-bd1a-1736865d6fc1)

### The workaround
`c3c compile src/main.c3 --obj-out temp --output-dir temp --libdir lib --lib littlec -o helloc`

**Comment by lerno** [comment]
Yes, it's likely the build directory isn't created at this point, which causes the issue.

**Comment by lerno** [comment]
This should work now.

**Comment by Caleb-o** [comment]
Awesome! This is working @lerno
